nix/overlay.nix

```diff
@@ -0,0 +1,111 @@
+{
+  config,
+  inputs,
+  ...
+}: {
+  flake.overlays.default = final: _: let
+    inherit
+      (final)
+      # Libraries
+      lib
+      stdenv
+      # Builders
+      buildPackages
+      fetchurl
+      fetchNpmDeps
+      makeWrapper
+      writeShellScriptBin
+      # Packages
+      cmake
+      deno
+      gtk3
+      libxkbcommon
+      libGL
+      xorg
+      nodejs
+      openssl
+      pkg-config
+      protobuf
+      wayland
+      yq
+      ;
+    inherit (buildPackages.npmHooks.override {inherit nodejs;}) npmConfigHook;
+    inherit (lib) concatStringsSep getExe' makeBinPath makeLibraryPath optional optionalString;
+    inherit (stdenv.hostPlatform) isLinux rust system;
+    # Borrowed from other packages in nixpkgs https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20RUSTY_V8_ARCHIVE&type=code
+    buildRustyV8Url = version: target: "https://github.com/denoland/rusty_v8/releases/download/v${version}/librusty_v8_release_${target}.a.gz";
+    fetchRustyV8 = version: hashes:
+      fetchurl {
+        name = "librusty_v8-${version}";
+        url = buildRustyV8Url version rust.rustcTarget;
+        hash = hashes.${system};
+        meta.version = version;
+        meta.sourceProvenance = [lib.sourceTypes.binaryNativeCode];
+      };
+    fetchRustyV8Hashes = writeShellScriptBin "fetch-librusty_v8-hashes.sh" ''
+      version=$(${getExe' yq "tomlq"} '.package | map(select(.name == "v8")) | .[0].version' --raw-output ${inputs.self}/Cargo.lock)
+      echo "Update ./nix/overlay.nix as follows:"
+      echo "RUSTY_V8_ARCHIVE = fetchRustyV8 \"$version\" {"
+      for system in ${concatStringsSep " " config.systems}; do
+          target=$(nix eval --raw nixpkgs#legacyPackages.$system.stdenv.hostPlatform.rust.rustcTarget)
+          hash=$(nix-prefetch-url --print-path ${buildRustyV8Url "$version" "$target"} | tail -n 1 | xargs nix hash file)
+          echo "  $system = \"$hash\";"
+      done
+      echo "};"
+    '';
+    pname = "gauntlet";
+    version = "v11";
```

Comment:
Would it be possible to move all versions and hashes to single place?

Reply:
Done